libraries/AP_BoardConfig/AP_BoardConfig_CAN.cpp

```diff
@@ -157,6 +162,38 @@ void AP_BoardConfig_CAN::px4_setup_canbus(void)
         }
     }
 }
+#elif CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+void AP_BoardConfig_CAN::linux_setup_canbus(void) {
+    for (uint8_t i = 0; i < MAX_NUMBER_OF_CAN_INTERFACES; i++) {
+    
+        uint8_t drv_num = _var_info_can[i]._driver_number;
+
+        if (drv_num != 0 && drv_num <= MAX_NUMBER_OF_CAN_DRIVERS) {
+            if (hal.can_mgr[drv_num - 1] == nullptr) {
+                const_cast <AP_HAL::HAL&> (hal).can_mgr[drv_num - 1] = new Linux::CANManager;
+            }
+            
+            if (hal.can_mgr[drv_num - 1] != nullptr) {
+                _var_info_can_protocol[i]._uavcan = new AP_UAVCAN;
+                if (_var_info_can_protocol[i]._uavcan != nullptr) {
+                    AP_Param::load_object_from_eeprom(_var_info_can_protocol[i]._uavcan, AP_UAVCAN::var_info);
+                
+                    hal.can_mgr[i]->set_UAVCAN(_var_info_can_protocol[i]._uavcan);
+                    _var_info_can_protocol[i]._uavcan->set_parent_can_mgr(hal.can_mgr[i]); 
+                
+                    if (!hal.can_mgr[i]->begin(_var_info_can[i]._can_bitrate, i)) {
+                        hal.console->printf("Failed to initialize can_mgr\n");
```

Comment:
would be nice if the failure showed the index, like the HAL_BOARD_PX4 one does

Reply:
as @EShamaev points out, it would be nice if we merged linux_setup_canbus() and px4_setup_canbus() into a single function setup_canbus()

Reply:
@Programmer74 would you like me to implement those changes for you and present a branch?